ipa-core/src/protocol/context/dzkp_validator.rs

```diff
@@ -399,25 +333,20 @@ impl UnverifiedValuesStore {
     /// insert `segments` for `segments` that are multiples of 256
     ///
     /// ## Panics
-    /// Panics when segment is not a multiple of 256.
+    /// Panics when segment is not a multiple of 256 or is out of bounds.
     fn insert_segment_large(&mut self, length: usize, position_vec: usize, segment: &Segment) {
         let length_in_entries = length >> BIT_ARRAY_SHIFT;
         for i in 0..length_in_entries {
-            let entry_option = &mut self.vec[position_vec + i];
-            // make sure we don't overwrite values
-            debug_assert!(entry_option.is_none());
-            *entry_option = Some(
-                UnverifiedValues::new(
-                    &segment.x_left.0[256 * i..256 * (i + 1)],
-                    &segment.x_right.0[256 * i..256 * (i + 1)],
-                    &segment.y_left.0[256 * i..256 * (i + 1)],
-                    &segment.y_right.0[256 * i..256 * (i + 1)],
-                    &segment.prss_left.0[256 * i..256 * (i + 1)],
-                    &segment.prss_right.0[256 * i..256 * (i + 1)],
-                    &segment.z_right.0[256 * i..256 * (i + 1)],
-                )
-                .unwrap(),
-            );
+            self.vec[position_vec + i] = UnverifiedValues::new(
```

Comment:
Wait, why are we calling `UnverifiedValues::new`? Didn't we already pre-allocate all of them and set them to zeroes?

Here's the code above that I have in mind:
```
vec: vec![
    UnverifiedValues::default();
    (amount_of_multiplications * multiplication_bit_size + 255) >> BIT_ARRAY_SHIFT
],
```

It seems like we should not need to do anything but over-write some bits. But I might be missing something or confused.